### PR TITLE
feat: re orgnize public API

### PR DESCRIPTION
**Pull request by kazuk**
## Issue number and link

Fixes: #158

## Describe your changes

- mod `api` into public 
- refactor `use` 
  - not to use `super` `self`, shoud be use `crate`
  - `use` ordinal `std` external crates, `crate::`
- file level ordinal , `mod` definition to `use` definition
- re-export high_level_rs members from api
- re-export low_level_rs members from api
- move low_level_rs::engine_mutex::EngineMutex to stream_engine
- move low_level_rs members to pipeline
- rename SpringPipelineHL to SpringPipeline
- rename SpringRowHL to SpringRow

## Checklist before requesting a review

- [x] I follow the [Semantic Pull Requests](https://github.com/zeke/semantic-pull-requests) rules (bugfix/feature)
- [x] I specified links to related issues (must: bugfix, want: feature)
- [x] I have performed a self-review of my code (bugfix/feature)
- [x] I have added thorough tests (bugfix/feature)
- [x] I have edited `## [Unreleased]` section in `CHANGELOG.md` following [keep a changelog](https://keepachangelog.com/en/1.0.0/) syntax (bugfix/feature)
- [x] I {made/will make} a related pull request for [documentation repo](https://github.com/SpringQL/SpringQL.github.io) (feature)

### CODE REVIEW

File: springql-core/src/lib.md [hunk 1]
```diff
@@ -160,6 +160,7 @@ fn main() {
                 NAME 'q_avg_by_symbol'
             );", ).unwrap();
 
+#   return ();
```
Comment (kazuk) [inline springql-core/src/lib.md:163]: fix: https://github.com/SpringQL/SpringQL/issues/168

File: springql-core/src/api/low_level_rs/spring_config.rs [hunk 2]
```diff
@@ -75,9 +75,9 @@ pub fn spring_config_default() -> SpringConfig {
 ///
 /// # Failures
 ///
-/// - [SpringError::InvalidConfig](crate::error::SpringError::InvalidConfig) when:
+/// - [SpringError::InvalidConfig](crate::api::error::SpringError::InvalidConfig) when:
 ///   - `overwrite_config_toml` includes invalid key and/or value.
-/// - [SpringError::InvalidFormat](crate::error::SpringError::InvalidFormat) when:
+/// - [SpringError::InvalidFormat](crate::api::error::SpringError::InvalidFormat) when:
 ///   - `overwrite_config_toml` is not valid as TOML.
 pub fn spring_config_toml(overwrite_config_toml: &str) -> Result<SpringConfig> {
```
Comment (laysakura) [inline springql-core/src/api/low_level_rs/spring_config.rs:82]: I guess we can remove this function (and write wrapper codes calling `SpringConfig::new(overwrite_config_toml)` in SpringQL-client-c/src/lib.rs ).

File: springql-core/src/api/low_level_rs/engine_mutex.rs [hunk 3]
```diff
@@ -1,34 +1 @@
 // This file is part of https://github.com/SpringQL/SpringQL which is licensed under MIT OR Apache-2.0. See file LICENSE-MIT or LICENSE-APACHE for full license details.
```
Comment (laysakura) [inline springql-core/src/api/low_level_rs/engine_mutex.rs:1]: Remove this file.

File: springql-core/src/api/low_level_rs.rs [hunk 4]
```diff
@@ -4,201 +4,6 @@
 //!
 //! C API and high-level Rust API are provided separately.
 
-mod engine_mutex;
-mod spring_config;
+pub mod spring_config;
```
Comment (laysakura) [inline springql-core/src/api/low_level_rs.rs:7]: We maybe able to totally remove this `low_level_rs` mod.

And then, we won't need to say "high-level" nor "low-level" anywhere.

**Review by laysakura — CHANGES_REQUESTED** [review]
I appreciate your diligent work!

Just one thing: I would like to totally remove "high-level" and "low-level" category from APIs and documentations.

I guess we can remove `low_level` mod in springql-core crate in this PR.

**Review by laysakura — APPROVED** [review]
👏

**Pull request merged**